DisruptorUnity3d/Assets/RingBuffer.cs

```diff
@@ -81,19 +81,19 @@ public bool TryDequeue(out T obj)
         /// <param name="item"></param>
         public void Enqueue(T item)
         {
-            var next = _producerCursor.ReadFullFence() + 1;
+            var next = _producerCursor.ReadAcquireFence() + 1;
```

Comment:
The problem is here. A read acquire must be used to make sure that the read from the array (later on) occurs after this read.